components/layout_2020/dom_traversal.rs

```diff
@@ -17,7 +17,7 @@ use style::properties::ComputedValues;
 use style::selector_parser::PseudoElement;
 
 #[derive(Clone, Copy)]
-pub(super) enum WhichPseudoElement {
+pub enum WhichPseudoElement {
```

Comment:
Are all of these newly-`pub` items used in another crate? I wonder if could move the crate boundaries somehow to make the API surface smaller. Maybe add some wrapper structs with private fields, for things that are exposed?

In any case this doesn’t need to block this PR.

Reply:
The trait `NodeExt` needs to be public, and thus any item exposed by the trait needs to be public too, so `WhichPseudoElement`, among other types, needs to be public. I could wrap LayoutBox into a struct but I'm not sure that it is *that* useful to do so.